### DISCUSSION TITLE
Clearing added text from the plot window

### DISCUSSION THREAD

**Discussion by SterlingButters** [pyvista/pyvista#5284] [Q&A] [answered]
First time poster

Looking to have the 3D coordinates of a mouse-click displayed in the plot window. Right now, I have the following:

```
class PrintMouseCallback:
    def __init__(self, plotter):
        self.plotter = plotter

    def __call__(self):
        print(self.plotter.picked_point)
        self.plotter.add_text(str(self.plotter.picked_point))

sphere = pv.Sphere()

p = pv.Plotter()

callback = PrintMouseCallback(p)
p.add_key_event('t', callback)

sphere_actor = p.add_mesh(sphere) 
p.enable_point_picking(pickable_window=True)
p.pickable_actors = [sphere_actor] 
p.show()
```

Which indeed, adds the point as text in the plot window, however this is in addition to last added text. Ideally, I wouldn't need the `t` key event but just trying to get something working right now. Thanks in advance

**Comment by SterlingButters** [comment]
Ok I sort of got a workaround but seems less than ideal - especially because it changes the mesh appearance after the second callback invocation:

```
class PrintMouseCallback:
    def __init__(self, plotter):
        self.plotter = plotter
        self.last = None

    def __call__(self, coord):
        if self.last is not None:
            actors = {k: v for k, v in self.plotter.actors.items() if v != self.last}
            self.plotter.clear()
            for actor, content in actors.items():
                self.plotter.add_actor(content)

        self.last = self.plotter.add_text(str(coord))

sphere = pv.Sphere()
p = pv.Plotter()
sphere_actor = p.add_mesh(sphere, opacity=0.5) 

callback = PrintMouseCallback(p)
p.enable_point_picking(callback,
                       left_clicking=False,
                       color='black',
                       show_message='')
p.show_grid(font_family='courier', bold=False, font_size=15) 

p.show()
```

**Reply by SterlingButters** [reply]
Ok this just isn't practical since it is relatively expensive to clear all the actors and then add them back. Hopefully someone can respond with a better alternative

**Reply by SterlingButters** [reply, marked as answer]
Ok this is much better:

```
class PrintMouseCallback:
    def __init__(self, plotter):
        self.plotter = plotter
        self.last = None

    def __call__(self, coord):
        if self.last is not None:
            for actor, content in self.plotter.actors.items():
                if content == self.last:
                    content.SetVisibility(False)

        self.last = self.plotter.add_text(str(coord))
```

**Comment by edabor** [comment]
HI ! Glad you found a solution !

You could also use the `name` parameters of the `add_text` method, which is specifically designed for your usage.
From the [docs](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.add_text.html):

`
The name for the added actor so that it can be easily updated. If an actor of this name already exists in the rendering window, it will be replaced by the new actor.
`

This is really usefull and is also available for most `add_*` Plotter methods.

**Reply by SterlingButters** [reply]
Yup that is definitely the optimal solution I was looking for - thanks!
